workitem/workitem.go

```diff
@@ -53,3 +57,66 @@ func (wi WorkItem) GetLastModified() time.Time {
 	log.Debug(nil, map[string]interface{}{"wi_id": wi.ID}, "Last modified value: %v", lastModified)
 	return *lastModified
 }
+
+// ChangeSet derives a changeset between this workitem and a given workitem.
+func (wi WorkItem) ChangeSet(older convert.ChangeDetector) ([]convert.Change, error) {
+	olderWorkItem, ok := older.(WorkItem)
+	if !ok {
+		return nil, errors.New("Other entity is not a WorkItem: " + reflect.TypeOf(older).String())
+	}
+	if wi.ID != olderWorkItem.ID {
+		return nil, errors.New("Other entity has not the same ID: " + olderWorkItem.ID.String())
+	}
+	changes := []convert.Change{}
+	// CAUTION: we're only supporting changes to the system.state and to the
+	// board position relationship for now. If we need to support more
+	// attribute changes, this has to be added here. This will be likely
+	// necessary when adding new Actions.
+	// compare system.state
+	if wi.Fields["system.state"] != olderWorkItem.Fields["system.state"] {
+		changes = append(changes, convert.Change{
+			AttributeName: "system.state",
+			NewValue:      wi.Fields["system.state"],
+			OldValue:      olderWorkItem.Fields["system.state"],
+		})
+	}
+	// compare system.boardcolumns
+	// this field looks like this:
+	// system.boardcolumns": ["43f9e838-3b4b-45e8-85eb-dd402e8324b5", "69699af8-cb28-4b90-b829-24c1aad12797"]
+	if wi.Fields["system.boardcolumns"] == nil && olderWorkItem.Fields["system.boardcolumns"] == nil {
+		return changes, nil
+	}
+	if wi.Fields["system.boardcolumns"] == nil || olderWorkItem.Fields["system.boardcolumns"] == nil {
+		changes = append(changes, convert.Change{
+			AttributeName: "system.boardcolumns",
+			NewValue:      wi.Fields["system.state"],
+			OldValue:      olderWorkItem.Fields["system.state"],
+		})
+		return changes, nil
+	}
+	bcThis, ok := wi.Fields["system.boardcolumns"].([]string)
+	bcOlder, ok := olderWorkItem.Fields["system.boardcolumns"].([]string)
```

Comment:
The `ok` here overwrites the ok from above which is never checked!

Reply:
In fact, none of the `ok`s seems to be checked.

Reply:
Outdated. Already fixed.